### PR TITLE
fix outside of array bounds exception to view a submission

### PR DESCRIPTION
**Pull request by huangjosh**

### CODE REVIEW

**Comment by weiqingtoh** [comment]
Thanks for this @huangjosh - I'm not sure if we necessarily need to guard against empty questions, because we have a [validation for published assessments to have at least 1 question](https://github.com/Coursemology/coursemology2/blob/master/app/models/course/assessment.rb#L176-L178). 

That said, it is good practice to have this to prevent crashes in the submissions page.

File: app/helpers/course/assessment/submission/submissions_autograded_helper.rb [hunk 1]
```diff
@@ -20,7 +20,8 @@ def next_unanswered_question
   # The step that current user is on.
   def current_step
     @current_step ||= begin
-      @assessment.questions.index(@current_question) + 1
+      idx = @assessment.questions.index(@current_question)
```
Comment (weiqingtoh) [inline app/helpers/course/assessment/submission/submissions_autograded_helper.rb]: Can shorten this to `@current_question ? @assessment.questions.index(@current_question) + 1 : nil`

File: app/views/course/assessment/submission/submissions/_autograded.html.slim [hunk 2]
```diff
@@ -10,7 +10,8 @@ div.answer-panel
     = f.error_notification
     = hidden_field_tag :step, current_step
     div.answer-content
-      = render partial: @current_question, suffix: 'question'
+      - if @current_question.present?
```
Comment (weiqingtoh) [inline app/views/course/assessment/submission/submissions/_autograded.html.slim]: `if @current_question` will work, since that checks for `nil`.

**Comment by huangjosh** [comment]
@allenwq It is a corner case. If a student submitted his answer. After that admin deleted all the questions for that assessment, the admin can still view the submission the student has previously submitted. In this case, the system will throw exception.

**Comment by weiqingtoh** [comment]
If that is the case, additionally, can we also add in a before_destroy callback to check if destroying the question would result in the assessment not having any questions? If I remember correctly, this is not thread safe either, so keeping this PR would helpful in that situation.

**Comment by huangjosh** [comment]
@weiqingtoh What's the expected behaviour when we want to destroy the last question? Do we just prevent the user from destroying the last question if the assessment is published or the any students have submitted their answers to the questions? 
All the rest errors in your comments are fixed.

**Comment by weiqingtoh** [comment]
> @weiqingtoh What's the expected behaviour when we want to destroy the last question? Do we just prevent the user from destroying the last question if the assessment is published or the any students have submitted their answers to the questions?

Ideal would be: 
 - When user goes to assessment page and tries to delete the question, if students have submitted their answers, we should have a confirmation dialog to say that students' answer data will be lost.
 - If user sees the dialog and still proceeds to delete the question, we can check if that is the last question. 
 - If it is the last question, prevent the user from deleting it, and surface the error along the lines of "published assessments must have at least 1 question. To delete this question, please unpublish the assessment first".
